addons/binding/org.openhab.binding.co2signal/src/main/java/org/openhab/binding/co2signal/internal/json/CO2SignalJsonResponse.java

```diff
@@ -0,0 +1,39 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.co2signal.internal.json;
+
+/**
+ * The {@link CO2SignalJsonResponse} is the Java class used to map the JSON
+ * response to the co2signal.com request.
+ *
+ * @author Jens Viebig - Initial contribution
+ */
+public class CO2SignalJsonResponse {
+
+    private String status;
+
+    private String countryCode;
+
+    private CO2SignalJsonData data;
+
+    public CO2SignalJsonResponse() {
```

Comment:
You don't have to add this construct, imho its overly verbose.
If there is no constructor Java will automatically add a default constructor (public with no-arguments).

Reply:
Disagree. Gson encourages to add a no-args constructor. It seems to be useful for reflection

Reply:
It only requires you to do so if you also have another no-no-args constructor.

Reply:
Sorry, still disagree on this one. While it might work without the construtor, use of an implicit constructor is still encouraged for gson: 

See:
* https://github.com/google/gson/blob/master/UserGuide.md#writing-an-instance-creator
> While deserializing an Object, Gson needs to create a default instance of the class. Well-behaved classes that are meant for serialization and deserialization should have a no-argument constructor.
*  https://stackoverflow.com/questions/18645050/is-default-no-args-constructor-mandatory-for-gson

Reply:
That's not how it works. If you have no constructor at all you will get a default constructor by definition. See the java language specification: http://docs.oracle.com/javase/specs/jls/se7/html/jls-8.html#jls-8.8.9 
If you add at least 1 constructor with arguments you won't get a default constructor and that is the case gson is referring too. In that case you do need to add a default constructor with no arguments to make gson play nice.